pcap/net_parse.go

```diff
@@ -94,26 +94,27 @@ func (p *NetworkTrafficParser) ParseFromInterface(interfaceName, bpfFilter strin
 
 	go func() {
 		ticker := time.NewTicker(streamTimeout / 4)
+		defer ticker.Stop()
 
-		defer func() {
-			ticker.Stop()
-
-			// Flushes and closes all remaining connections. This should trigger all
-			// parsers to hit EOF and return. This call will block until the parsers
-			// have returned because tcpStream.ReassemblyComplete waits for
-			// parsers.
-			assembler.FlushAll()
-
-			// Signal caller that we're done here.
-			close(out)
-		}()
+		// Signal caller that we're done on exit
+		defer close(out)
```

Comment:
NB: This reverses the order in which `ticker.Stop()` and `close(out)` are executed.